src/ui_kit/src/file/mod.rs

```diff
@@ -58,8 +59,23 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
                                 ContextItem {
                                     icon: Shape::DocumentArrowDown,
                                     onpressed: move |_| {
-                                        // TODO(Files): Add download function here
-                                        eprintln!("Download item");
+                                        let file_storage = cx.props.storage.clone();
+                                        let file_name = &*file_name_complete_ref.read();
+                                        let file_extension = cx.props.kind.clone();
+                                        cx.spawn({
+                                            to_owned![file_storage, file_name, file_extension];
+                                            async move {
+                                                let file_path_buff = match FileDialog::new().set_directory(".").set_file_name(&file_name).add_filter("", &[&file_extension]).save_file() {
+                                                    Some(path) => path,
+                                                    None => return,
+                                                };
+                                                let file_path_str = file_path_buff.to_string_lossy().to_string();
+                                                match file_storage.get(&file_name, &file_path_str).await {
```

Comment:
Similar to `Constellation::put_stream`, we could use `Constellation::get_stream` to get a stream and write that to the location (you can refer to the example https://github.com/Satellite-im/Warp/blob/main/extensions/warp-fs-ipfs/examples/ipfs-persistent.rs#L154-L169.)

Note: If that function is used, it only returns a stream of `Result<Vec<u8>, Error>` (which is why in the example you see `while let Some(Ok(data))`) so any progression needing to be done would to be done by the UI (since it doesnt output the same as `Constellation::put_stream`)